provider.go

```diff
@@ -140,33 +142,38 @@ func New(options ...ProviderHandler) (*WasmcloudProvider, error) {
 		putTargetLinkFunc: func(InterfaceLinkDefinition) error { return nil },
 		delSourceLinkFunc: func(InterfaceLinkDefinition) error { return nil },
 		delTargetLinkFunc: func(InterfaceLinkDefinition) error { return nil },
+		sourceLinkArr:     sourceLinks,
+		targetLinkArr:     targetLinks,
 
 		sourceLinks: make(map[string]InterfaceLinkDefinition, len(sourceLinks)),
 		targetLinks: make(map[string]InterfaceLinkDefinition, len(targetLinks)),
 	}
+	return provider, nil
+}
 
+// setup link events for provider
+func ConfigureLinkEvents(provider *WasmcloudProvider, options ...ProviderHandler) error {
```

Comment:
Can we call this function as a part of the `Start()` function, before we subscribe to NATS? That way we can establish all of the links automatically for the user, just after the `New()` function so there's no race condition

```suggestion
// setup link events for provider
func establishLinks(provider *WasmcloudProvider, options ...ProviderHandler) error {
```

Reply:
So you want to modify the start function to accept options?
i.e.,
```
func (wp *WasmcloudProvider) Start(options ... ProviderHandler) error {

}
```

Reply:
Done . Please review.